GSA/digitalgov.gov
# Update Hugo

## Summary

Update Hugo version.

### Preview

Images should load properly throughout the site.

- [Homepage](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/jm-bugfix-time-strings/)
- [Communities](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/jm-bugfix-time-strings/communities/)
- [Services](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/jm-bugfix-time-strings/services/)
- [delivering-digital-first-public-experience](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/jm-bugfix-time-strings/resources/delivering-digital-first-public-experience/) (Check for missing icons)

### Solution

1. Updated `.hugo-version`.
2. Removed dates from external content [f9cf991].
3. Fixed broken dates to get proper values [09c100f].

## Major changes

### Fixed dates in content

Incorrect dates in content were causing meta tags to **not** display. 

**How to test** 
1. Visit: `/resources/how-test-websites-for-accessibility`, this content has a publish date but doesn't display on live. 
2. Open devtools  and search for `article:published_time`.
3. Confirm its missing on `live` & displays in preview.

---

### Relative URLs

> [!IMPORTANT]  
> Using a leading slash could cause regressions in CloudGov previews.

To make relative links work (images/internal links) work on Cloud Page previews we need to remove the leading `/` from URLs.

**Example**

```diff
<!-- Notice leading `/` is gone →
- <img src="{{- "/img/digit-100.svg" | relURL -}}" />
+ <img src="{{- "img/digit-100.svg" | relURL -}}" />
```

This follows [Hugo best practices](https://gohugo.io/functions/urls/relurl/#:~:text=The%20last%20three%20examples%20are%20not%20desirable%20in%20most%20situations.%20As%20a%20best%20practice%2C%20never%20include%20a%20leading%20slash%20when%20using%20this%20function.):

>As a best practice, never include a leading slash when using this function.


## Additional information


~- Citizen Science logo in services `/services` is broken, existing bug captured in Trello.~

Fixed in #7367, thanks @ToniBonittoGSA!

### How To Test

1. Pull down branch.
2. Update local hugo to match the one specified in `.hugo-version`.
3. Run `npm start`.
4. Confirm there aren't any build issues.

---

### Dev Checklist

- [ ] Site builds correctly
- [ ] Zero broken links to relative URLs
- [ ] Images load correctly
- [ ] Zero issues in Circleci builds
- [ ] Any changes in functionality are documented in pull request
